### PR TITLE
QP-3041 bypass version checks, rebuild thrift files

### PR DESCRIPTION
**Pull request by MahidaANokia**
Addresses [QP-3041](https://deepfield.atlassian.net/browse/QP-3041)

(The files to look at are _thrift_api.py and hiveserver2.py, the rest were modified by recompiling)

Impyla uses .py files compiled by thrift. At some point impyla switched to using thrift files directly instead of the compiled .py files due to incompatibility issues, but at a performance cost. Since then, thrift has been updated and can once again generate .py files that are usable, but we also need to bypass python version checks in a few places to force it to use these files the "old way".

We'll also need to add a conda recipe and make the appropriate systems changes to switch to using this fork, up to now we've been using the original cloudera version of impyla

## How to Test
1. On a vlab, clone the deepfield impyla repo with `git clone git@github.com:deepfield/impyla.git`
2. Checkout this branch
3. First, check query performance before the change. In an ipython shell:
```
import deepy.ibis.util

with deepy.ibis.util.default_connection() as con:
    %time cur = con.raw_sql('select * from deepfield_traffic_step10.traffic limit 100000', results=True)
    %time cur.fetchall(columnar=True)
```
You should get something like:
```
CPU times: user 28 ms, sys: 4 ms, total: 32 ms
Wall time: 4.07 s
CPU times: user 39 s, sys: 456 ms, total: 39.5 s
Wall time: 40.1 s
```
4. "Install" these changes by copying the impala directory from impyla to the site-packes directory: `sudo cp -r impala/ /pipedream/local/venv/deepfield-env/lib/python3.7/site-packages/`
5. Repeat step 3, this time the user+sys time should go down by an order of magnitude, something like:
```
CPU times: user 20 ms, sys: 0 ns, total: 20 ms
Wall time: 3.78 s
CPU times: user 1.38 s, sys: 328 ms, total: 1.7 s
Wall time: 2.42 s
```

### CODE REVIEW

File: impala/_thrift_gen/TCLIService/TCLIService.py [hunk 1]
```diff
@@ -1,3832 +1,3834 @@
...
 
-from thrift.Thrift import TType, TMessageType, TException, TApplicationException
+from thrift.Thrift import TType, TMessageType, TFrozenDict, TException, TApplicationException
+from thrift.protocol.TProtocol import TProtocolException
+from thrift.TRecursive import fix_spec
+
+import sys
 import logging
-from ttypes import *
+from .ttypes import *
```
Comment (maxmzkrdf) [inline impala/_thrift_gen/TCLIService/TCLIService.py:15]: not going to look through all the thrift files, but this is the specific line I was looking for to make sure we have a python3 compatible file.

**Comment by maxmzkrdf** [comment]
wondering if we want to merge into master or into a release branch? Merging into master is probably fine, but I think we'll probably want to cut a tag off a release?

**Comment by MahidaANokia** [comment]
@maxmzkrdf i was thinking we would just merge this into master of this repo, create the conda recipe, and then change versions accordingly in anaconda-deps in the systems versions we want it in so older releases would keep using the unmodified impyla from conda. is that different from what you had in mind?

**Comment by maxmzkrdf** [comment]
So first thing of note is that your changes are based off of v0.16.2. I think that's fine for deepfield master, but we probably want to stick to v0.15.0 for 4.4 and 5.0 to reduce the number of changes we're making on an already QA'd release branch. Because of this, we'll want to be able to track the changes we make to v0.15.0 and v0.16.2 separately. In the past, I have found it easiest to make release branches for the tags and then apply the changes to the release branches. In this case, you would have a v0.15.0 and a v0.16.2 release branches. You would apply your changes to both of those. After applying the changes, you would cut two tags v0.15.0.0 and v0.16.2.0. You can also apply your changes to master, but the Deepfield impyla master should probably track the cloudera impyla master and not the tag v0.16.2. A release branch would likely be better for tracking that tag.

In the end, the branching structure would look something like this

```
A - B - C - D - E
 \       \       \
  F       G       H
```

A would be v0.15.0 and F would be your changes and also the tag v0.15.0.0. C would be v0.16.2 and G would be your changes and also the tag v0.16.2.0. E would be cloudera master and H would be your changes and also deepfield impyla master.

similarly documented in our hadoop fork
https://deepfield.atlassian.net/wiki/spaces/~maxmzkr/pages/1183744139/Hadoop+fork?search_id=2ddc71b0-ddd9-475b-9ba0-8002828950ae

and similar to what we did for our ibis fork
https://github.com/deepfield/ibis

At the end of the day, it doesn't matter too much to me what branching structure we chose, but we should have it documented. What ever we chose should be able to handle any additional patches we may end up needing to make and also support patches on multiple different impyla releases.

**Comment by MahidaANokia** [comment]
Ok, that makes sense to me. How does this play with conda recipes though? I know in meta.yaml you can set a tag or branch, would we have to make separate recipes for the different branches?

**Comment by maxmzkrdf** [comment]
Ya, you would have to make multiple. I don't know if we version our conda recipes currently.

**Comment by MahidaANokia** [comment]
Ok, i've made two additional branches, `df-v0.15.0` and `df-v0.16.2` based off their respective tags, and i've rebased this to df-v0.16.2. Let me know if you're ok with this and i'll merge it, and then get the conda recipes set up.

**Pull request merged**